src/devices/OneWire/OneWireController.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Iot.Device.OneWire
+{
+    /// <summary>
+    /// Represents a 1-wire controller.
+    /// </summary>
+    public static partial class OneWireController
+    {
+        /// <summary>
+        /// Enumerate all 1-wire busses in the system.
+        /// </summary>
+        /// <returns>A list of discovered busses.</returns>
+        public static IEnumerable<OneWireBus> EnumerateBuses()
```

Comment:
I got mixed feeling on creating OneWireBus/OneWireDevice objects when enumerating. Especially I'd expect OneWireDevice to open the file in the constructor and keep it open until it is disposed.

I think I'd go with enumerating names/ids of the buses instead of the object and for enumerating devices either return a tuple or create a OneWireDeviceInfo struct which contains bus id/name and device name/id

Reply:
We are talking one small object (bus instance) extra in normal case. I can make it a struct BusInfo. But I don't see any big gain over keeping it easy to understand/use. Device should probably be an object since user will hold on to it for longer. So I prefer them to be similar Bus+Device instad of BusInfo+Device.
And I don't like holding a system resource longer than necessary. IDispose is easy to miss and when a read is async and take quite some time there is no need to cache the file handle. I go for ease of use over micro optimization.

Reply:
for Buses I'd return string. I think it's generally better to hold system resources for things you're gonna use for longer period of time (thermometer)

Reply:
Why, then you would have to make all bus operations static and take a bus id as input.

Reply:
Or you want the user to create a OneWireBus from a string/busid?

Reply:
I actually didn't notice there is no string constructor but yes, I'd prefer bus was created from string and this API enumerated names.

Reply:
Out of curiosity: does the bus id stay the same when rebooting OS?

Reply:
yes, it is the Linux device name